packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -0,0 +1,990 @@
+/********************************************************************************
+ * Copyright (C) 2020 Arm and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable no-null/no-null, @typescript-eslint/no-explicit-any */
+
+import * as React from 'react';
+import { injectable, inject, postConstruct } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { DisposableCollection, Disposable } from '@theia/core/lib/common/disposable';
+import { TreeWidget, TreeNode, TreeProps, NodeProps, TreeModel, CompositeTreeNode, SelectableTreeNode, ExpandableTreeNode,
+    TREE_NODE_SEGMENT_GROW_CLASS } from '@theia/core/lib/browser/tree';
+import { MenuModelRegistry, ActionMenuNode, CompositeMenuNode, MenuPath } from '@theia/core/lib/common/menu';
+import { ScmResourceGroup, ScmResource, ScmResourceDecorations } from './scm-provider';
+import { ScmService } from './scm-service';
+import { CommandRegistry } from '@theia/core/lib/common/command';
+import { ScmRepository } from './scm-repository';
+import { ContextMenuRenderer, LabelProvider, CorePreferences, DiffUris} from '@theia/core/lib/browser';
+import { ScmContextKeyService } from './scm-context-key-service';
+import { EditorWidget } from '@theia/editor/lib/browser';
+import { EditorManager, DiffNavigatorProvider } from '@theia/editor/lib/browser';
+import { FileStat } from '@theia/filesystem/lib/common';
+import { IconThemeService } from '@theia/core/lib/browser/icon-theme-service';
+
+export interface FileChangeGroupRoot extends CompositeTreeNode {
```

Comment:
Please make sure that all types have `Scm` prefix.

Reply:
'Scm' prefixed to all interfaces here (moved to scm-tree-model).